src/util/mod.rs

```diff
@@ -554,6 +556,14 @@ pub fn is_zero_duration(d: &Duration) -> bool {
     d.as_secs() == 0 && d.subsec_nanos() == 0
 }
 
+/// A hash function for u64, return value is less than 'upper'.
+#[inline]
+pub fn hash_u64(i: u64, upper: u64) -> u64 {
```

Comment:
IMO, hash function only means getting the hash value, not includes calculating modulus.

Reply:
Maybe we can change the name.

Reply:
OK, I will change the function name.